File: src/main/java/election/ElectionClient.java
package election;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;

public class ElectionClient {

    public static void main(String[] args) {
        ManagedChannel channel = ManagedChannelBuilder.forAddress("localhost", 50051)
                .usePlaintext()
                .build();

        ElectionServiceGrpc.ElectionServiceBlockingStub stub = ElectionServiceGrpc.newBlockingStub(channel);

        Election.ElectionResponse response = stub.getElectionData(Election.ElectionRequest.newBuilder()
                .setElectionId("2024-Presidential")  // Beispiel-ID
                .build());

        System.out.println("Election ID: " + response.getElectionId());
        System.out.println("Candidate: " + response.getCandidateName());
        System.out.println("Votes: " + response.getVotes());

        channel.shutdown();
    }
}

File: election_client.py
import grpc
import election_pb2
import election_pb2_grpc

def run():
    channel = grpc.insecure_channel('localhost:50051')
    stub = election_pb2_grpc.ElectionServiceStub(channel)

    # Anfrage mit einer Beispiel-Wahl-ID
    request = election_pb2.ElectionRequest(electionId="2024-Presidential")

    try:
        response = stub.getElectionData(request)
        print("Election ID:", response.electionId)
        for candidate in response.candidates:
            print("Candidate:", candidate.candidateName, "\nVotes:", candidate.votes, "\n")
    except grpc.RpcError as e:
        print(f"RPC failed: {e.code()}: {e.details()}")

if __name__ == "__main__":
    run()
